### PR TITLE
Add missing types for WindowsCodec

### LINKED ISSUE #1735: Codegen for `IPropertyBag2` and its friends
This is a follow-up to #1674.

It seems that we still need to introduce the support for `IPropertyBag2`, `tagPROPVARIANT` and `tagPROPBAG2`, otherwise it can be hard to use. For example, https://learn.microsoft.com/en-us/windows/win32/wic/-wic-creating-encoder#tiff-encoding-example

Those types are located in oaidl.h and ocidl.h.

Is there any way to only include the types we actually need without introducing the whole bunch of unnecessary dependencies?

_Originally posted by @hez2010 in https://github.com/dotnet/Silk.NET/issues/1674#issuecomment-1741700823_

### PR DESCRIPTION
**Pull request by hez2010**
# Summary of the PR
Add missing types for WIC, typically `IPropertyBag`, `PROPVARIANT` and etc.
Also added `IDispatch` and its friends.

# Related issues, Discord discussions, or proposals
Closes #1735

### CODE REVIEW

File: generator.json [hunk 1]
```diff
@@ -1817,16 +2021,16 @@
...
-                    "tagPROPVARIANT": "void",
-                    "tagPROPBAG2": "void",
-                    "HPALETTE__": "void",
-                    "HBITMAP__": "void",
-                    "HICON__": "void",
+                    "tagPROPVARIANT": "Silk.NET.Core.Win32Extras.PropVariant",
+                    "tagPROPBAG2": "Silk.NET.Core.Win32Extras.PropBag2",
+                    "HPALETTE__": "IntPtr",
+                    "HBITMAP__": "IntPtr",
+                    "HICON__": "IntPtr",
```
Comment (Perksey) [inline generator.json:2033]: What's the reason for this change? A `HICON__` is the opaque struct pointed to by a `HICON` (that is, a `HICON` is a `HICON__*` where `HICON__` is empty). We have sometimes used `IntPtr` in place of pointers to types like these, but not in place of the opaque pointee. I appreciate that `void*` isn't the best dev ex, but I'm not sure why this is the change.

Can we revert this?

File: src/Core/Silk.NET.Core.Win32Extras/Structs/IPrintDocumentPackageStatusEvent.gen.cs [hunk 2]
```diff
@@ -261,190 +264,190 @@ public IPrintDocumentPackageStatusEvent
         }
 
         /// <summary>To be documented.</summary>
-        public readonly unsafe int Invoke(int dispIdMember, Guid* riid, uint lcid, ushort wFlags, DispatchParams* pDispParams, Variant* pVarResult, ExceptionInfo* pExcepInfo, uint* puArgErr)
+        public readonly unsafe int Invoke(int dispIdMember, Guid* riid, uint lcid, ushort wFlags, DispatchParams* pDispParams, Variant* pVarResult, ExcepInfo* pExcepInfo, uint* puArgErr)
```
Comment (Perksey) [inline src/Core/Silk.NET.Core.Win32Extras/Structs/IPrintDocumentPackageStatusEvent.gen.cs:267]: Why has this renamed?

File: src/Core/Silk.NET.Core.Win32Extras/Structs/IPropertyNotifySink.gen.cs [hunk 3]
```diff
@@ -16,19 +16,19 @@
 
 namespace Silk.NET.Core.Win32Extras
 {
-    [Guid("947990de-cc28-11d2-a0f7-00805f858fb1")]
-    [NativeName("Name", "IDummyHICONIncluder")]
-    public unsafe partial struct IDummyHICONIncluder : IComVtbl<IDummyHICONIncluder>, IComVtbl<Silk.NET.Core.Native.IUnknown>
```
Comment (Perksey) [inline src/Core/Silk.NET.Core.Win32Extras/Structs/IPropertyNotifySink.gen.cs:21]: What happened to `IDummyHICONIncluder`?

**Review by Perksey — CHANGES_REQUESTED** [review]

File: src/Core/Silk.NET.Core.Win32Extras/Structs/IPrintDocumentPackageStatusEvent.gen.cs (hunk 2 above)
Comment (hez2010) [reply src/Core/Silk.NET.Core.Win32Extras/Structs/IPrintDocumentPackageStatusEvent.gen.cs]: This one is due to `tagEXCEPINFO` being generated again, I have excluded it in win32extras.

File: src/Core/Silk.NET.Core.Win32Extras/Structs/IPropertyNotifySink.gen.cs (hunk 3 above)
Comment (hez2010) [reply src/Core/Silk.NET.Core.Win32Extras/Structs/IPropertyNotifySink.gen.cs]: Removed by mistake. Resolved.

File: generator.json (hunk 1 above)
Comment (hez2010) [reply generator.json]: Done.

**Comment by hez2010** [comment]
PTAL

**Review by Perksey — APPROVED** [review]
